### PR TITLE
#166265907 check authors added to book

### PR DESCRIPTION
**Pull request by PelumiAlesh**
authorID added to a book should be checked if it exist in the database

[Finishes]
### What does this PR do?
Check if a author exist in the database when adding book.
#### Description of Task to be completed?
When a user adds authorID to a book being published :
>- This PR would check if any author with the specified ID exists in the database.
>- If it does not exist, it would throw an error
#### How should this be manually tested?
After cloning this repo :
>- Run `npm i` on the command line
>- Run `npm run dev` on the command line
>- Make a `POST` request to `localhost:8080/api/v1/Books`
#### Any background context you want to provide?
null
#### What are the relevant pivotal tracker stories?
 #166265907   
#### Screenshots (if appropriate)
Screenshot 2019-06-06 at 12 42 02 PM
#### Questions:

### CODE REVIEW

**Review by emp-daisy — APPROVED** [review]

File: server/middleware/validation.js [hunk 1]
```diff
@@ -91,7 +91,14 @@ const validate = {
       .isEmpty({ ignore_whitespace: true })
       .withMessage('AuthorID can not be left empty: Please input amount')
       .isInt()
-      .withMessage('AuthorID is not valid integer: Please input a valid authorID'),
+      .withMessage('AuthorID is not valid integer: Please input a valid authorID')
+      .custom( async (id) => {
+        const isExist = await util.checkAuthorID(id);
+        if (!isExist) {
+          throw new Error('No author with the specified ID was found')
```
Comment (chuxmykel) [inline server/middleware/validation.js:98]: Why are you throwing and is this error caught somewhere?

Comment (emp-daisy) [reply server/middleware/validation.js:98]: https://express-validator.github.io/docs/custom-validators-sanitizers.html

Comment (PelumiAlesh) [reply server/middleware/validation.js:98]: That is how to write express validator custom function. it would catch the error its self.

**Review by chuxmykel — APPROVED** [review]
Looks good then.

**Review by OvieMudi — APPROVED** [review]

**Pull request merged**